src/js/tools/superselect.js

```diff
@@ -74,25 +75,27 @@ define(function (require, exports, module) {
                         UI.pointerPropagationMode.ALPHA_PROPAGATE_WITH_NOTIFY);
                 });
         } else {
-            var currentLayers = currentDocument.layers.selected,
-                currentLayer = currentLayers.first();
-            
-            if (currentLayer.vectorMaskEnabled && !currentLayer.vectorMaskEmpty) {
-                return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing())
-                    .bind(this)
-                    .then(function () {
-                        var pointerPolicy = new PointerEventPolicy(UI.policyAction.ALWAYS_PROPAGATE,
-                            OS.eventKind.LEFT_MOUSE_DOWN);
-                        return this.transfer(policy.addPointerPolicies, [pointerPolicy]);
-                    })
-                    .then(function () {
-                        // We are not transfering here, because we activly want to end the use of our locks
-                        this.flux.actions.tools.enterPathModalState();
-                    });
-            } else {
-                var pointerPolicy = new PointerEventPolicy(UI.policyAction.ALWAYS_PROPAGATE,
-                    OS.eventKind.LEFT_MOUSE_DOWN);
-                return this.transfer(policy.addPointerPolicies, [pointerPolicy]);
+            var currentLayers = currentDocument.layers.selected;
+
+            if (currentLayers) {
+                var currentLayer = currentLayers.first();
+
+                if (currentLayer && currentLayer.vectorMaskEnabled) {
```

Comment:
If you do the above emptiness check, you can eliminate the first conjunct above because there will always be a first selected layer given that the selection is non-empty.